app/hoc/withAuth/tests/index.test.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { MemoryRouter, Switch, Route } from 'react-router';
+import { containerWithStore } from 'helpers/store';
+import { fakeSession, fakeEmptySession } from 'mocks/fakeSession';
+import { paths } from 'helpers/routes';
+import withAuth from '../';
+
+describe('withAuth HOC', () => {
+  let state;
+
+  const WrappedComponent = () => null;
+  const Home = () => null;
+
+  const renderComponent = () => {
+    const EnhancedComponent = () => containerWithStore(withAuth(WrappedComponent), state);
+
+    return mount(
+      <MemoryRouter initialEntries={ ['/protected'] } >
+        <Switch>
+          <Route
+            exact
+            path={ paths.home() }
+            component={ Home }
+          />
+          <Route path="/protected" component={ EnhancedComponent } />
+        </Switch>
+      </MemoryRouter>
+    );
+  };
+
+  context('when authenticated', () => {
+    beforeEach(() => {
+      state = {
+        session: fakeSession
+      };
+    });
+
+    it('renders WrappedComponent', () => {
+      const component = renderComponent().find(WrappedComponent);
```

Comment:
how would this test possibly fail?

Reply:
This test will fail in case if `state.session.loggedIn === false`. (see [this condition](https://github.com/fs/react-base/pull/101/files/82b239565aef943ff1b064caac27f05931456bf1#diff-f333a45c609aa553463eb3d6b275b1e8R14))

Reply:
so state.session thing is under arrange section of test. and we have full control of that. what's the point of having that? if i go to withAuth and delete random string or change anything would this test fail?

Reply:
Yup, you can change anything in `render` method of HOC and test will fail.